src/backend/web/graphql/index.js

```diff
@@ -94,7 +122,27 @@ module.exports.resolvers = {
         const last = first + prPage < numOFPosts ? first + prPage : numOFPosts;
         const postIds = await getPosts(first, last);
 
-        return Promise.all(postIds.map(id => Post.byId(id)));
+        const result = await Promise.all(postIds.map(id => Post.byId(id)));
+
+        if (filter) {
+          // check if author name is equal to what we're searching for
+          if (filter.author) {
+            return result.filter(post => post.author === filter.author);
+          }
+          // check if post's date is between two provided dates
+          if (filter.fromDate || filter.toDate) {
+            const fromDate = filter.fromDate ? filter.fromDate : new Date();
+            const toDate = filter.toDate ? filter.toDate : new Date();
+            const datePostIds = await getPostsByDate(fromDate, toDate);
+            return result.filter(post => datePostIds.includes(post.id));
+          }
+          // check if url is equal to what we're searching for
+          if (filter.url) {
+            return result.filter(post => post.url === filter.url);
+          }
+        }
+
+        return result;
```

Comment:
I don't know why the results are being the way they are. 

It might also be I'm not entirely sure of how the flow should be at the moment.
How I'm thinking it should work

1. We get the page results, so say we're getting the first 30 results of page 1.
2. We filter those 30 results by the date filter now

or should we ...
1. Get the results with the filter
2. Apply pagination on the filtered results

Playing around with the filter, I have no idea why the results are so different:
for example with 
```
{
  getPosts(page: 1, perPage: 10, filter: {fromDate: "2020-01-01"}){
    author
    published
    title
  }
}
```
![image](https://user-images.githubusercontent.com/18711727/74008065-f5995e80-494d-11ea-9798-ee88f5d8f2bd.png)

but if I change perPage to say 20
```
{
  getPosts(page: 1, perPage: 20, filter: {fromDate: "2020-01-01"}){
    author
    published
    title
  }
}
```
![image](https://user-images.githubusercontent.com/18711727/74008154-2da0a180-494e-11ea-86ed-84bc25a9f733.png)

The results seem like they're going backwards..?

Reply:
I might be that you get counts based on not filtering, but then sometimes filter (by date).  The posts and count should only get processed once you know about the filters you are using.